src/core/mapping/base_mapper.cc

```diff
@@ -477,16 +477,14 @@ void BaseMapper::map_task(const MapperContext ctx,
   auto default_option            = options.front();
   auto generate_default_mappings = [&](auto& stores, bool exact) {
     for (auto& store : stores) {
+      auto mapping = StoreMapping::default_mapping(store, default_option, exact);
       if (store.is_future()) {
-        auto option  = default_option == StoreTarget::FBMEM ? StoreTarget::ZCMEM : default_option;
```

Comment:
Note the change in #518 wasn't actually necessary, as the target will be overriden in the following `map_futures` function anyway.